src/Tools/Test Framework/Test Runner/src/TestRunnerMgt.Codeunit.al

```diff
@@ -37,30 +40,34 @@ codeunit 130454 "Test Runner - Mgt"
         TestMethodLine.SetRange("Line Type", TestMethodLine."Line Type"::Codeunit);
 
         ALCodeCoverageMgt.Initialize(TestMethodLine."Test Suite");
-        OnRunTestSuite(TestMethodLine);
+        this.OnRunTestSuite(TestMethodLine);
 
+        BackupTestFilter := TestMethodLine.GetFilter("Line No.");
         if TestMethodLine.FindSet() then
             repeat
-                OnBeforeCodeunitRun(TestMethodLine);
+                this.CurrentTestFilter := TestSuiteMgt.GetLineNoFilterForTestCodeunit(TestMethodLine);
+                TestMethodLine.SetFilter("Line No.", this.CurrentTestFilter);
+                this.OnBeforeCodeunitRun(TestMethodLine);
                 CODEUNIT.Run(TestMethodLine."Test Codeunit");
+                this.OnAfterCodeunitRun(TestMethodLine);
+                TestMethodLine.SetFilter("Line No.", BackupTestFilter);
                 TestMethodLine.Find();
-                OnAfterCodeunitRun(TestMethodLine);
             until TestMethodLine.Next() = 0;
 
-        OnAfterRunTestSuite(TestMethodLine);
+        this.OnAfterRunTestSuite(TestMethodLine);
     end;
 
     /// This method is called when the caller needs to run a test codeunit but do not want to log results or the caller has 
-    /// an alternateway to log the results. Currently, this is used by the Performance Toolkit
+    /// an alternately to log the results. Currently, this is used by the Performance Toolkit
```

Comment:
There is also an issue in line 60 that I cannot mark on github.com.
/// This method is called when the caller needs to run a test codeunit but do**es** not want to log results or the caller has 
///  an alternat**ive** way to log the results. Currently, this is used by the Performance Toolkit